### PR TITLE
fix vanilla test status on CI

### PR DESCRIPTION
**Pull request by aykut-bozkurt**
Fixes vanilla test status on CI.

- Because of the make command used for vanilla tests, test status is always shown as success on CI. As a fix, I added `&& false` at the end of the copying diff file to make the command fail when check-vanilla fails.
```make
check-vanilla: all
	$(pg_regress_multi_check) --vanillatest || (cp $(vanilla_diffs_file) $(citus_abs_srcdir)/regression.diffs && false)
```

- I also fixed some vanilla tests that fails due to recently added clock related operators shown up at some queries.

### CODE REVIEW

File: src/test/regress/Makefile
Comment (JelteF) [inline src/test/regress/Makefile:194]: I'm assuming you checked that CI fails after this change and without the others?

**Review by JelteF — APPROVED** [review]
LGTM, let's wait for review from @tejeswarm, though since some failures were introduced by his clock related changes.

File: src/test/regress/Makefile
Comment (aykut-bozkurt) [reply src/test/regress/Makefile:194]: yep.

**Comment by tejeswarm** [comment]
> * I also fixed some vanilla tests that fails due to recently added clock related operators shown up at some queries.

I don't have enough context, I remember fixing other upgrade tests as we added new clock operators, but these seem to Citus source code changes and not tests. @aykut-bozkurt can we please sync offline

**Review by tejeswarm — APPROVED** [review]
Did we recently add access-methods too? `pg_amop`

**Comment by aykut-bozkurt** [comment]
> Did we recently add access-methods too? `pg_amop`

No, but one of the PG test, namely oidjoins, fails due to foreign key violation if we do not hide pg_amop as well.

i.e. `(pg_amop,{amopfamily},pg_opfamily,{oid},f,f)`

**Pull request merged**
